Backpack/Carousel/Classes/BPKCarousel.swift

```diff
@@ -0,0 +1,183 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import Foundation
+
+public final class BPKCarousel: UIView {
+    
+    public var images: [UIView] {
+        didSet {
+            collectionViewDatasource.render(with: images)
+            pageIndicator.totalIndicators = images.count
+        }
+    }
+    
+    public var currentImage: Int? {
+        didSet {
+            guard let currentImage = currentImage else { return }
+            scrollTo(row: currentImage, animated: true)
+            updateSelected(index: currentImage)
+        }
+    }
+    
+    public var onImageChanged: ((Int) -> Void)?
```

Comment:
Have we considered a delegate pattern instead of closures for this component? I think it'd work better for the UIKit version.

Properties like this (or the `images`) property work better for SwiftUI, because of it's reactive nature, but in UIKit it forces users to do non ideal code.

Reply:
Yes delegate will be used